File: Code/Components/DanpheEMR.ServerModel/LabModels/LabTestModel.cs

using DanpheEMR.ServerModel.LabModels;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DanpheEMR.ServerModel
{
    public class LabTestModel
    {
        [Key]
        public Int64 LabTestId { get; set; }
        public string LabTestCode { get; set; }
        public int LabSequence { get; set; }
        public string ProcedureCode { get; set; }
        public string LabTestName { get; set; }
        public string LabTestSynonym { get; set; }
        public string LabTestSpecimen { get; set; }
        public string LabTestSpecimenSource { get; set; }

        public string LOINC { get; set; }
        public int? ReportTemplateId { get; set; }
        public bool? IsValidForReporting { get; set; }
        public string Description { get; set; }
        public int? DisplaySequence { get; set; }
        public string RunNumberType { get; set; }//Anish:26Sept'18       

        public DateTime? CreatedOn { get; set; }
        public int? CreatedBy { get; set; }
        public bool? IsActive { get; set; }

        public bool? HasNegativeResults { get; set; }
        public string NegativeResultText { get; set; }
        public int? LabTestCategoryId { get; set; }

        public virtual LabReportTemplateModel LabReportTemplate { get; set; }


        //ashim: 06Sep2018
        public string ReportingName { get; set; }

        //anish: 9 Sept 2018
        public string Interpretation { get; set; }

        public int? ModifiedBy { get; set; }
        public DateTime? ModifiedOn { get; set; }

        // below property 'IsSelected' is not the part of databae model, 
        //but we need some boolean field to bind with checkbox so kept it.<dharmendra:13jan'17> 
        [NotMapped]
        public bool? IsSelected { get; set; }
        [NotMapped]
        public bool? IsPreference { get; set; }
        //suraj: 24th sepptember 2018
        [NotMapped]
        public bool? IsTaxApplicable { get; set; }
        [NotMapped]
        public List<LabTestJSONComponentModel> LabTestComponentsJSON { get; set; }
        [NotMapped]
        public List<LabTestComponentMapModel> LabTestComponentMap { get; set; }


        //NBB-29 Jan 2018- searched but not any reference found for this property in our application        
        //and MyProperty is not a valid column in db also, so now i'm applying [NotMapped] decorator
        [NotMapped]
        public int? MyProperty { get; set; }

        [NotMapped]
        public int? ServiceDepartmentId { get; set; }

        [NotMapped]
        public string TemplateType { get; set; }


        //public ICollection<LabTestJSONComponentModel> LabTestComponents { get; set; }
    }
}

File: Code/Websites/DanpheEMR/wwwroot/DanpheApp/src/app/labs/shared/lab-test.model.ts
import { LabReportTemplateModel } from './lab-report-template.model';
//import { LabTestGroup } from './lab-testgroup.model';
import {
    FormBuilder,
    FormGroup,
    Validators
} from '@angular/forms';
import * as moment from 'moment/moment';
import { ENUM_DateTimeFormat } from '../../shared/shared-enums';
import { LabTestComponentMap } from '../lab-settings/shared/lab-test-component-map.model';
import { LabComponentModel } from './lab-component-json.model';
export class LabTest {
    public LabTestId: number = 0;
    public LabSequence: number = 0;
    public LabTestCode: string = null;
    public ProcedureCode: string = null;
    public Description: string = null;
    public LabTestSynonym: string = null;
    public LabTestName: string = null;
    public LabTestSpecimen: string = null;
    public LabTestSpecimenSource: string = null;
    public LabTestComponentsJSON: Array<LabComponentModel> = [];
    public LabTestComponentMap: Array<LabTestComponentMap> = [];
    public LOINC: string = null;
    public ReportTemplateId: number = 0;
    public IsSelected: boolean = false;
    public IsPreference: boolean = false;
    public IsValidForReporting: boolean = true;
    public CreatedOn: string = null;
    public CreatedBy: number = 0;
    public ModifiedOn: string = null;
    public ModifiedBy: number = null;
    public IsActive: boolean = true;
    public HasNegativeResults: boolean = false;
    public NegativeResultText: string = null;

    public ServiceDepartmentId: number = null;

    public LabReportTemplateModel: LabReportTemplateModel = null;

    public DisplaySequence: number = 1000;
    public RunNumberType: string = 'normal';

    public LabTestValidator: FormGroup = null;

    //ashim: 06Sep2018
    public ReportingName: string = null;
    public Interpretation: string = null;
    public IsTaxApplicable: boolean = false;

    public ReportTemplateName: string = null;

    public TemplateType: string = null;
    public LabTestCategoryId: number = 0;
    public GroupName: string;

    public SmsApplicable: boolean = false;

    public IsOutsourceTest: boolean = false;
    public DefaultOutsourceVendorId: number = 0;
    public IsLISApplicable: boolean = false;
    constructor() {
        this.CreatedOn = moment().format(ENUM_DateTimeFormat.Year_Month_Day_Hour_Minute);
        var _formBuilder = new FormBuilder();
        this.LabTestValidator = _formBuilder.group({
            'LabTestName': ['', Validators.compose([Validators.required])]
        });
    }

    public IsDirty(fieldname): boolean {
        if (fieldname == undefined) {
            return this.LabTestValidator.dirty;
        }
        else {
            return this.LabTestValidator.controls[fieldname].dirty;
        }

    }

    public IsValid(): boolean { if (this.LabTestValidator.valid) { return true; } else { return false; } }


    public IsValidCheck(fieldname, validator): boolean {

        if (fieldname == undefined) {
            return this.LabTestValidator.valid;
        }
        else {
            return !(this.LabTestValidator.hasError(validator, fieldname));
        }
    }


}
